service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -193,14 +217,28 @@ public boolean updateWorkspace(
             "UPDATE workspace SET %s WHERE workspace_id = :workspace_id",
             DbUtils.setColumnsClause(params, "properties"));
 
-    int rowsAffected = jdbcTemplate.update(sql, params);
-    boolean updated = rowsAffected > 0;
+    int rowsAffected;
+    try {
+      rowsAffected = jdbcTemplate.update(sql, params);
+    } catch (DuplicateKeyException e) {
+      // Workspace with user_facing_id already exists.
```

Comment:
This could be a duplicate of the guid as well.

Reply:
This is `updateWorkspace()`, guid can't be updated. (There's no id field in UpdateWorkspaceRequestBody.)